clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -399,6 +399,14 @@ mlir::Type CIRGenTypes::convertType(QualType type) {
     break;
   }
 
+  case Type::ExtVector:
+  case Type::Vector: {
+    const VectorType *vec = cast<VectorType>(ty);
+    const mlir::Type elemTy = convertTypeForMem(vec->getElementType());
```

Comment:
I'm not sure `convertTypeForMem` is what we want here. Specifically, I don't think we want a vector of bools to end up as a vector of i8. I tried this with the incubator, but it seems vectors of bool are not yet implemented there.